sentry/src/main/java/io/sentry/SentryOptions.java

```diff
@@ -217,6 +218,29 @@
   /** Enable the Java to NDK Scope sync */
   private boolean enableScopeSync;
 
+  /**
+   * Enables loading additional options from external locations like {@code sentry.properties} file
+   * or environment variables, system properties.
+   */
+  private boolean enableExternalConfiguration;
+
+  /**
+   * Creates {@link SentryOptions} from properties provided by a {@link PropertiesProvider}.
+   *
+   * @param propertiesProvider the properties provider
+   * @return the sentry options
+   */
+  public static @NotNull SentryOptions from(final @NotNull PropertiesProvider propertiesProvider) {
+    final SentryOptions options = new SentryOptions();
+    options.setDsn(propertiesProvider.getProperty("dsn"));
+    options.setEnvironment(propertiesProvider.getProperty("environment"));
+    options.setRelease(propertiesProvider.getProperty("release"));
+    options.setDist(propertiesProvider.getProperty("dist"));
+    options.setServerName(propertiesProvider.getProperty("servername"));
+    // TODO: which properties should be set?
```

Comment:
This question still remains. Should we set all possible properties or just the most important ones?

Reply:
IMO the most important ones, we can add more later, the thing is, if we don't add them all, we need to "tell how to" do it manually to the missing ones.